numpy/lib/tests/test_function_base.py

```diff
@@ -4110,6 +4110,13 @@ def test_quantile_weights_raises_negative_weights(self):
         with pytest.raises(ValueError, match="Weights must be non-negative"):
             np.quantile(y, 0.5, weights=w, method="inverted_cdf")
 
+    def test_quantile_weights_raises_all_zeros_weights(self):
+        y = [1, 2]
+        w = [0, 0]
+        with pytest.raises(ValueError, match="Weights must contain non-zero value."):
+            np.quantile(y, 0.5, weights=w, method="inverted_cdf")
+
+
```

Comment:
> Could someone please provide guidance on the reasons for these failures and suggest steps to address them?

You can reproduce the linter failure observed in CI locally by running `spin lint` on your branch. Purging the extra blank line it complains about will allow `spin lint` to pass locally and in CI. I checked the full test suite locally on this branch and it seems "ok."

The Qemu job failure in CI is clearly unrelated so don't worry about that.

For bug fixes, we usually prefix commits and titles of PRs with `BUG:`. You can find the list of prefixes in our docs.

Note for other NumPy devs--gh-28594 is very similar to this.